### PR TITLE
Group Advanced Settings - Change Path

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Updated the group settings page to separate updates to path and parent namespace within the advanced settings section.
Fixes the first half of #75.

## Screenshots or screen recordings
image

## How to set up and validate locally

How to test the new form:
1. Navigate to the new group page
2. Enter a group name.
3. Verify path is slugified.
4. Hit the `Create group` button.
5. Verify the group was created.

How to test the edit form:
1. Navigate to a group settings page.
2. Update path.
3. Hit the `Change path` button.
4. Verify the path changed. 

Note: It might be a good idea to test both groups and subgroups. 

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the PR acceptance checklist for this PR.

### CODE REVIEW

**Comment by ksierks** [comment]
> As per our conversation last week, could you please update the structure/styling of the Groups Settings page to match the structure/styling of the Project Settings page that was recently merged in.
> 
> e.g. image

Please see 1583ce658ef83dc19b21d348bdf154d431135a7d.

**Comment by joshsadam** [comment]
Just a quick comment on the overall appearance:



1.  There are different gaps under all the buttons, I prefer the gap under button 1.
2.  Buttons have different heights, this might be fixed in my updated button PR #131 and could be caused by something in the class definition that currently exists (i.e. I don't think it is something that you did, but needs to be addressed).  Maybe we should hold on this PR until the button on is merged in to main. 
    1.  1 is 40px
    2.  2 is 42px
    3.  3 is 36px

**Comment by joshsadam** [comment]
Just a quick comment on the overall appearance:



1.  There are different gaps under all the buttons, I prefer the gap under button 1.
2.  Buttons have different heights, this might be fixed in my updated button PR #131 and could be caused by something in the class definition that currently exists (i.e. I don't think it is something that you did, but needs to be addressed).  Maybe we should hold on this PR until the button on is merged in to main. 
    1.  1 is 40px
    2.  2 is 42px
    3.  3 is 36px

**Review by ericenns — CHANGES_REQUESTED** [review]
Reviewing #75 the changes here don't quite match what was requested.

Specifically title in the card should be updated to `Change group URL`, along with the submitting button text and the parent namespace should not be allowed to be changed as that is covered in the transferring group which you have a separate PR for.

See image below for what it looks like in gitlab, which our redesign is based off of.

MicrosoftTeams-image (6)

Also, after updating the look please make sure to update the screenshots in the PR description as we want to make sure these are always up to date with the latest changes.

**Review by joshsadam — CHANGES_REQUESTED** [review]
One small change here.  I will take a look again once the other issues are addressed.  Thanks @ksierks

**Comment by ksierks** [comment]
> Reviewing #75 the changes here don't quite match what was requested.
> 
> Specifically title in the card should be updated to `Change group URL`, along with the submitting button text and the parent namespace should not be allowed to be changed as that is covered in the transferring group which you have a separate PR for.
> 
> See image below for what it looks like in gitlab, which our redesign is based off of.
> 
> MicrosoftTeams-image (6)
> 
> Also, after updating the look please make sure to update the screenshots in the PR description as we want to make sure these are always up to date with the latest changes.

I was so focused on moving the existing change path while not breaking subgroups. That makes more sense. Working on fixing it.

**Comment by ksierks** [comment]
> > Reviewing #75 the changes here don't quite match what was requested.
> > Specifically title in the card should be updated to `Change group URL`, along with the submitting button text and the parent namespace should not be allowed to be changed as that is covered in the transferring group which you have a separate PR for.
> > See image below for what it looks like in gitlab, which our redesign is based off of.
> > MicrosoftTeams-image (6)
> > Also, after updating the look please make sure to update the screenshots in the PR description as we want to make sure these are always up to date with the latest changes.
> 
> I was so focused on moving the existing change path while not breaking subgroups. That makes more sense. Working on fixing it.

I'm unsure how exact you want it to look like GitLab's Change URL. I left the form label for accessibility and the default button.  Most importantly, the parent group drop-down is removed and I also added the tooltip and form field title. Also, the screenshot above has been updated. Hope this looks okay.

Screenshot from 2023-07-10 13-35-55

File: app/views/groups/_path_form_fields.html.erb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          class="
+            inline-flex
+            items-center
+            px-3
+            py-2.5
+            text-sm
+            text-gray-600
+            bg-gray-200
+            dark:bg-gray-800/50
+            dark:text-gray-200
```
[the remark this answers is not in the document]
Comment (ksierks) [reply app/views/groups/_path_form_fields.html.erb]: I think this was quickly copied from _project_namespace_fields.erb

**Review by ericenns — CHANGES_REQUESTED** [review]
Just noting our conversation regarding a few more changes:

* change the submit button to red
* remove the label for `parent_id` and the hidden `parent_id` input
* reduce to single form-field with prefixed span showing root url, and move class to input called `prefixed` (following example set out in select.prefixed in group creation form)
* fix css width issue when input has errors
* fix wrong url for path when attempting to update group to existing path causes error
  * note: this issue is existing and was not introduced in this pr

**Review by ericenns — APPROVED** [review]
Looks good to me! Thanks for making all the changes

**Pull request merged**